### PR TITLE
Ansible role Group first working commit

### PR DESCRIPTION
**Pull request by eyenx**
##### SUMMARY
New ansible role for groups (additional groups, not created by ansible-role-users)

##### ANSIBLE VERSION

```
2.7.4
```

### CODE REVIEW

File: meta/.galaxy_install_info [hunk 1]
```diff
@@ -0,0 +1 @@
+{install_date: 'Tue Dec 11 09:09:26 2018', version: master}
```
Comment (karras) [inline meta/.galaxy_install_info]: What's this file?

File: meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  min_ansible_version: '2.0.0'
+  platforms:
+    - name: Debian
+      versions:
+        - wheezy
+        - jessie
+        - stretch
+    - name: Ubuntu
+      versions:
+        - trusty
```
Comment (karras) [inline meta/main.yml]: Add bionic

File: tasks/configuration.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+
+- name: create groups
+  group:
+    name: '{{ item.name }}'
+    system: '{{ item.system | default(omit) }}'
+    gid: '{{ item.gid | default(omit) }}'
+    state: '{{ item.state | default("present") }}'
+  tags:
```
Comment (karras) [inline tasks/configuration.yml]: We don't add  tags on task level, only in the main.yml.

File: tasks/installation.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: install groups related packages
+  package:
+    name: '{{ groups_packages }}'
+    state: present
+  register: groups_register_packages_installed
+  until: groups_register_packages_installed is succeeded
+  tags:
```
Comment (karras) [inline tasks/installation.yml]: We don't add tags on task level, only in the main.yml. What's up with that `until`?

File: vars/Debian.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+
```
Comment (karras) [inline vars/Debian.yml]: Add var description

File: vars/RedHat.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+
```
Comment (karras) [inline vars/RedHat.yml]: Add var description

**Review by karras — CHANGES_REQUESTED** [review]

File: tasks/installation.yml (hunk 4 above)
Comment (eyenx) [reply tasks/installation.yml]: Took this from ansible-role-users. Needs to be update there too, imho.

Until: See https://github.com/adfinis-sygroup/ansible-role-icinga2_web/pull/1/commits/b5e8983a1a363476bf1e6d008cc5eadf38bebaa4 same issue.

File: meta/.galaxy_install_info (hunk 1 above)
Comment (eyenx) [reply meta/.galaxy_install_info]: deleted.

File: vars/RedHat.yml (hunk 6 above)
Comment (eyenx) [reply vars/RedHat.yml]: done

File: vars/Debian.yml (hunk 5 above)
Comment (eyenx) [reply vars/Debian.yml]: done

File: tasks/installation.yml (hunk 4 above)
Comment (eyenx) [reply tasks/installation.yml]: fixed as suggested by @fujexo

File: tasks/configuration.yml (hunk 3 above)
Comment (eyenx) [reply tasks/configuration.yml]: removed

File: defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+# .. code-block:: YAML
+#
+#   groups_group:
+#     - name: wheel
+#       gid: 1000
+#       system: true
+#       state: present
+#
+# The default group list is empty.
+groups_group: []
```
Comment (fujexo) [inline defaults/main.yml]: Can you add an example?

**Review by fujexo — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 7 above)
Comment (eyenx) [reply defaults/main.yml]: isn't the example which is commented out enough?

Comment (fujexo) [reply defaults/main.yml]: Sorry man, I'm still blind. Should get more coffee before doing reviews

**Review by fujexo — APPROVED** [review]
LGTM

**Comment by eyenx** [comment]
status?

**Review by karras — APPROVED** [review]

**Comment by karras** [comment]
LGTM, a professional nitpicker could argue that the role doesn't remove groups as describe in the meta description.

**Comment by eyenx** [comment]
see https://github.com/adfinis-sygroup/ansible-role-groups/issues/3

**Pull request merged**